toolchain/parse/state.def

```diff
@@ -697,7 +697,7 @@ CARBON_PARSE_STATE_VARIANTS2(ParenCondition, If, While)
 //   (state done)
 CARBON_PARSE_STATE_VARIANTS2(ParenConditionFinish, If, While)
 
-// Handles the `(` of a parenthesized expression.
+// Handles the `(` of a parenthesis expression or tuple literal
```

Comment:
You're right that this isn't what I intended, but what I intended was to drop "or tuple literal", so that "parenthesis expression" (abbreviated as `ParenExpr` in code) is the term for any expression headed by `(`. And yes, I intentionally used the noun form rather than the verb form, by analogy with e.g. "brace expression". That avoids the problem @zygoloid pointed out, that it's weird to use the term "parenthesized expression" for a tuple literal, where the thing inside parentheses is not an expression.

I've rephrased this to specify that a parenthesis is a parenthesized single expression or a tuple literal.